src/jupyter_contrib_nbextensions/templates/toc2.tpl

```diff
@@ -16,7 +16,7 @@
 <script>
 $( document ).ready(function(){
             var cfg={'threshold':6,     // depth of toc (number of levels)
-             'number_sections':true,    // sections numbering
+             'number_sections': "{{nb['metadata']['toc']['number_sections']}}"=="True" ? true : false,  // sections numbering
```

Comment:
it would probably be more robust to use something like

```jinja2
'number_sections': {{ 'true' if nb.get('metadata', {}).get('toc', {}).get('number_sections', False) else 'false' }},
```